drivers/serial/uart_nrfx_uart.c

```diff
@@ -180,6 +237,8 @@ static int uart_nrfx_fifo_fill(struct device *dev,
 {
 	u8_t num_tx = 0;
 
+	device_busy_set(dev);
```

Comment:
The device_busy_set/clear just sets the flag. Regardless of number of set() call, single clear() clears busy status.
This means that we have to track HW and driver state in the software, as we have to distinguish busy driver (API during execution) and busy hardware (data being sent) in order to clear the busy flag only when both driver and hardware is idle.